src/fe-text/gui-readline.c

```diff
@@ -638,7 +654,11 @@ static gboolean paste_timeout(gpointer data)
 {
 	paste_was_bracketed_mode = paste_bracketed_mode;
 
-	if (paste_line_count == 0) {
+	/* number of lines after splitting extra-long messages */
+	int split_lines = paste_buffer->len / LINE_SPLIT_LIMIT;
+
+	/* Take into account the fact that a line may be split every LINE_SPLIT_LIMIT characters */
+	if (paste_line_count == 0 && split_lines <= paste_verify_line_count) {
```

Comment:
I'm not this is correct. try pasting a very long line without linebreaks with and without your patch